forui/test/src/widgets/accordion/accordion_controller_test.dart

```diff
@@ -0,0 +1,258 @@
+        ..addItem(3, animationControllers[3], animations[3], initiallyExpanded: false);
+      await controller.toggle(1);
+      expect(controller.controllers[1]?.animation.value, 100);
+      await controller.toggle(3);
+      expect(controller.controllers[0]?.animation.value, 0);
+    });
+
+    test('expand(...)', () async {
+      controller
+        ..addItem(0, animationControllers[0], animations[0], initiallyExpanded: false)
+        ..addItem(1, animationControllers[1], animations[1], initiallyExpanded: false)
+        ..addItem(2, animationControllers[2], animations[2], initiallyExpanded: false)
+        ..addItem(3, animationControllers[3], animations[3], initiallyExpanded: true);
+      await controller.expand(0);
+      expect(controller.expanded, {3, 0});
+      await controller.expand(0);
+      expect(count, 1);
+      await controller.expand(1);
+      await controller.expand(2);
+      expect(controller.expanded, {0, 1, 2});
+    });
+
+    test('collapse(...)', () async {
+      controller
+        ..addItem(0, animationControllers[0], animations[0], initiallyExpanded: false)
+        ..addItem(1, animationControllers[1], animations[1], initiallyExpanded: false)
+        ..addItem(2, animationControllers[2], animations[2], initiallyExpanded: false)
+        ..addItem(3, animationControllers[3], animations[3], initiallyExpanded: true);
+      await controller.collapse(3);
+      expect(controller.expanded.contains(3), true);
+      expect(count, 0);
+      await controller.expand(0);
+      await controller.collapse(3);
+      await controller.collapse(2);
+      expect(count, 2);
+      await controller.collapse(0);
+      expect(controller.expanded, {0});
+    });
+
+    test('validate(...)', () {
+      expect(controller.validate(0), false);
+      expect(controller.validate(1), true);
+      expect(controller.validate(2), true);
+      expect(controller.validate(3), true);
+      expect(controller.validate(4), false);
+    });
+  });
+
+  group('FAccordionController.radio', () {
+    late FAccordionController controller;
+    final List<AnimationController> animationControllers = [];
+    final List<Animation<int>> animations = [];
+    int count = 0;
+
+    setUp(() {
+      count = 0;
+      animations.clear();
+      animationControllers.clear();
+
+      for (int i = 0; i < 4; i++) {
+        animationControllers.add(MockAnimationController());
+        animations.add(Tween(begin: 0, end: 100).animate(animationControllers[i]));
+        when(animationControllers[i].forward()).thenAnswer((_) {
+          when(animationControllers[i].value).thenReturn(1.0);
+          return TickerFuture.complete();
+        });
+        when(animationControllers[i].reverse()).thenAnswer((_) {
+          when(animationControllers[i].value).thenReturn(0.0);
+          return TickerFuture.complete();
+        });
+      }
+
+      controller = FAccordionController.radio()
+        ..addListener(() {
+          count++;
+        });
+    });
```

Comment:
This logic seems like it's duplicated. Please extract them out. Same with `tearDown(...)`.